test/support/protocol_utils.rb

```diff
@@ -203,6 +203,53 @@ def assert_reattach
       end
     end
 
+    def assert_locals_result expected, frame_idx: 0
+      case ENV['RUBY_DEBUG_TEST_UI']
+      when 'vscode'
+        # get frameId
+        send_request 'stackTrace',
+                      threadId: 1,
+                      startFrame: 0,
+                      levels: 20
+        res = find_crt_dap_response
+        f_id = res.dig(:body, :stackFrames, frame_idx, :id)
+
+        # get variablesReference
+        send_request 'scopes', frameId: f_id
+        res = find_crt_dap_response
+        locals_scope = res.dig(:body, :scopes).find { |d| d[:presentationHint] == "locals" }
+        locals_reference = locals_scope[:variablesReference]
+
+        # get variables
+        send_request 'variables', variablesReference: locals_reference
+        res = find_crt_dap_response
+
+        expected.each do |exp|
+          if exp[:type] == "String"
+            exp[:value] = exp[:value].inspect
+          end
+        end
+
+        actual_locals = res.dig(:body, :variables).map { |loc| { name: loc[:name], value: loc[:value], type: loc[:type] } }
+      when 'chrome'
+        current_frame = @crt_frames.first
+        locals_scope = current_frame[:scopeChain].find { |f| f[:type] == "local" }
+        object_id = locals_scope.dig(:object, :objectId)
+        send_request "Runtime.getProperties", objectId: object_id
+        res = find_crt_cdp_response
+
+        actual_locals = res.dig(:result, :result).map do |loc|
+          type = loc.dig(:value, :className) || loc.dig(:value, :type).capitalize # TODO: sync this with get_ruby_type
+
+          { name: loc[:name], value: loc.dig(:value, :description), type: type }
+        end
+      end
+
+      failure_msg = FailureMessage.new{create_protocol_message "result:\n#{JSON.pretty_generate res}"}
+
+      assert_equal(expected, actual_locals, failure_msg)
```

Comment:
Is it intentional to check the lvar's order?
I'm not sure it should be checked or not.

Reply:
no it's not intentional. I've updated it.